src/native/corehost/fxr/fx_resolver.cpp

```diff
@@ -519,6 +574,11 @@ StatusCode fx_resolver_t::resolve_frameworks_for_app(
 
     assert(retry_count < Max_Framework_Resolve_Retries);
 
+    if (rc == StatusCode::Success)
+    {
+        rc = check_framework_references_arch_compatible(resolver.m_effective_fx_references);
```

Comment:
I think this needs to react to `fx_definitions` instead. The `m_effective_fx_references` will contain the version for which the app asked, not the version it will run on. For example if I run a 5.0 app with `--roll-forward Major` on a machine which only has 6.0, the `m_effective_fx_reference` will have `5.0`, but the `fx_definitions` will have `6.0`.

Since we can only run on actual resolved frameworks, the decision needs to be based on `fx_definitions` which are actual frameworks on disk.

In the above sample, even though the app is 5.0, it can't run on 5.0 since that's not present on the machine, so it will run on 6.0 and thus should run on ARM64. The current logic would try to run it on x64 (and fail, since the 6.0 won't exist for x64).